src/hydra/HydraAdmin.js

```diff
@@ -7,34 +7,57 @@ import restClient from './hydraClient';
 class HydraAdmin extends Component {
   static defaultProps = {
     apiDocumentationParser,
+    customRoutes: [],
     restClient,
   };
 
   static propTypes = {
     apiDocumentationParser: PropTypes.func,
+    customRoutes: PropTypes.array,
     entrypoint: PropTypes.string.isRequired,
     restClient: PropTypes.func,
   };
 
   state = {
     api: null,
+    customRoutes: [],
+    hasError: false,
+    loaded: false,
   };
 
   componentDidMount() {
-    this.props
-      .apiDocumentationParser(this.props.entrypoint)
-      .then(api => this.setState({api}));
+    this.props.apiDocumentationParser(this.props.entrypoint).then(
+      ({api, customRoutes = []}) =>
+        this.setState({
```

Comment:
this feels duplicated, is it possible to not duplicate it or it's not worth the complexity ?

Reply:
Instead of call `setState` directly, we could "normalize" the data:

```js
    this.props
      .apiDocumentationParser(this.props.entrypoint)
      .then(
        ({api, customRoutes = []}) => ({
          api,
          customRoutes,
          hasError: false,
          loaded: true,
        }),
        ({api, customRoutes = []}) => ({
          api,
          customRoutes,
          hasError: true,
          loaded: true,
        }),
      )
      .then(state => this.setState(state));
```